lib/moab/storage_object_validator.rb

```diff
@@ -157,7 +158,8 @@ def check_metadata_dir_files_only(version_path)
       errors = []
       metadata_dir_path = File.join(version_path, DATA_DIR, METADATA_DIR)
       errors << result_hash(NO_FILES_IN_METADATA_DIR, basename(version_path)) if directory_entries(metadata_dir_path).empty?
-      errors << result_hash(METADATA_SUB_DIRS_DETECTED, basename(version_path)) if contains_sub_dir?(metadata_dir_path)
+      metadata_sub_dir = contains_sub_dir?(metadata_dir_path)
```

Comment:
Not a blocker, but I like when methods that end in `?` return a truthy/falsey value, but here you're using the returned value as a string.